modules/network/src/integrationTest/java/org/apache/ignite/network/scalecube/ItScaleCubeNetworkMessagingTest.java

```diff
@@ -355,25 +357,44 @@ public void nodeCannotReuseOldId(TestInfo testInfo) throws Exception {
     }
 
     private void knockOutNode(String outcastName) throws InterruptedException {
-        CountDownLatch disappeared = new CountDownLatch(1);
+        CountDownLatch disappeared = new CountDownLatch(2);
 
-        testCluster.members.get(0).topologyService().addEventHandler(new TopologyEventHandler() {
+        TopologyEventHandler disappearListener = new TopologyEventHandler() {
             @Override
             public void onDisappeared(ClusterNode member) {
                 if (Objects.equals(member.name(), outcastName)) {
                     disappeared.countDown();
                 }
             }
-        });
+        };
 
-        testCluster.members.stream()
+        List<ClusterService> notOutcasts = testCluster.members.stream()
                 .filter(service -> !outcastName.equals(service.nodeName()))
-                .forEach(service -> {
-                    DefaultMessagingService messagingService = (DefaultMessagingService) service.messagingService();
-                    messagingService.dropMessages((recipientConsistentId, message) -> outcastName.equals(recipientConsistentId));
-                });
+                .collect(Collectors.toList());
 
+        notOutcasts.forEach(clusterService -> {
+            clusterService.topologyService().addEventHandler(disappearListener);
+        });
+
+        notOutcasts.forEach(service -> {
+            DefaultMessagingService messagingService = (DefaultMessagingService) service.messagingService();
+            messagingService.dropMessages((recipientConsistentId, message) -> outcastName.equals(recipientConsistentId));
+        });
+
+        // Wait until all nodes see disappearance of the outcast.
         assertTrue(disappeared.await(10, TimeUnit.SECONDS), "Node did not disappear in time");
+
+        DefaultMessagingService messagingService = (DefaultMessagingService) testCluster.members.stream()
+                .filter(service -> outcastName.equals(service.nodeName()))
+                .findFirst()
+                .get().messagingService();
+
+        ConnectionManager cm = messagingService.connectionManager();
+
+        // Forcefully close channels, so that on reanimation nodes will create new channels.
+        cm.channels().forEach((stringConnectorKey, nettySender) -> {
+            nettySender.close().awaitUninterruptibly();
```

Comment:
I wonder should we do this (close channels) in the production code when we understand that the node has left (or is going to leave)? Otherwise, it looks suspicious: maybe it's a bug?

Reply:
I'm not sure. We will close a channel if it was closed. In this case, we artificially drop messages, but the channel is still alive, that's why channel is not closed. In normal conditions, I expect channel to be closed automatically either due to a channel reset or a timeout